src/Abstractions/TaskOptions.cs

```diff
@@ -6,153 +6,61 @@ namespace Microsoft.DurableTask;
-    }
-
-    /// <summary>
-    /// Gets the retry policy that was configured for this <see cref="TaskOptions"/> instance.
-    /// </summary>
-    public RetryPolicy? RetryPolicy { get; }
+    /// <param name="policy">The policy to convert from.</param>
+    /// <returns>A <see cref="TaskOptions" /> built from the policy.</returns>
+    public static TaskOptions FromRetryPolicy(RetryPolicy policy) => new(policy);
 
     /// <summary>
-    /// Gets the cancellation token that was configured for this <see cref="TaskOptions"/> instance.
+    /// Returns a new <see cref="TaskOptions" /> from the provided <see cref="AsyncRetryHandler" />.
     /// </summary>
-    public CancellationToken CancellationToken { get; }
+    /// <param name="handler">The handler to convert from.</param>
+    /// <returns>A <see cref="TaskOptions" /> built from the handler.</returns>
+    public static TaskOptions FromRetryHandler(AsyncRetryHandler handler) => new(handler);
 
     /// <summary>
-    /// Gets the retry handler that was configured for this <see cref="TaskOptions"/> instance.
+    /// Returns a new <see cref="TaskOptions" /> from the provided <see cref="RetryHandler" />.
     /// </summary>
-    public AsyncRetryHandler? RetryHandler { get; }
-
-    /// <summary>
-    /// Convenience method from creating a <see cref="TaskOptions"/> object from a <see cref="RetryPolicy"/>.
-    /// </summary>
-    /// <param name="policy">The task retry policy to configure.</param>
-    /// <param name="cancellationToken">Optional cancellation token for canceling the task.</param>
-    /// <returns>Returns a newly created <see cref="TaskOptions"/> object.</returns>
-    public static TaskOptions FromRetryPolicy(RetryPolicy policy, CancellationToken cancellationToken = default)
+    /// <param name="handler">The handler to convert from.</param>
+    /// <returns>A <see cref="TaskOptions" /> built from the handler.</returns>
+    public static TaskOptions FromRetryHandler(RetryHandler handler)
     {
-        return CreateBuilder().WithRetryStrategy(policy).WithCancellationToken(cancellationToken).Build();
+        Check.NotNull(handler);
+        return FromRetryHandler(context => Task.FromResult(handler.Invoke(context)));
     }
 
     /// <summary>
-    /// Convenience method from creating a <see cref="TaskOptions"/> object from a <see cref="RetryHandler"/>.
+    /// Gets the instance ID, if available, for this options instance.
     /// </summary>
-    /// <param name="retryHandler">The task retry handler to configure.</param>
-    /// <param name="cancellationToken">Optional cancellation token for canceling the task.</param>
-    /// <returns>Returns a newly created <see cref="TaskOptions"/> object.</returns>
-    public static TaskOptions FromRetryHandler(RetryHandler retryHandler, CancellationToken cancellationToken = default)
+    /// <returns>The orchestration instance ID if available, <c>null</c> otherwise.</returns>
+    internal string? GetInstanceId()
```

Comment:
Might need to make this `public` for external implementors to consume.

Reply:
External implementors of what, specifically?

Reply:
Functions implementation of `TaskOrchestrationContext` is one example - it would want access to this. But I think I can address this with a refactor.

Reply:
I opted to remove this. Implementors should determine their own way to test for instanceId.